torch/distributions/constraints.py

```diff
@@ -241,6 +242,23 @@ def __repr__(self):
         return fmt_string
 
 
+class _OpenInterval(Constraint):
```

Comment:
`torch.distributions.transforms` does not support a notion of openness or closedness. If we add this initial partial implementation it will set precedent for an unmanageable proliferation of edge cases and new support.

Can you clarify your motivation here and try to think of alternatives?